### PR TITLE
Fix on branch master, consumer invoke timeout cause NPE.

### PR DESCRIPTION
**Pull request by BazookaW**
issue no #8586

### CODE REVIEW

**Review by changfubai — APPROVED** [review]
LGTM.

File: dubbo-rpc/dubbo-rpc-dubbo/src/main/java/org/apache/dubbo/rpc/protocol/dubbo/DecodeableRpcResult.java [hunk 1]
```diff
@@ -118,11 +118,13 @@ public Object decode(Channel channel, InputStream input) throws IOException {
     public void decode() throws Exception {
         if (!hasDecoded && channel != null && inputStream != null) {
             try {
-                if (ConfigurationUtils.getSystemConfiguration().getBoolean(SERIALIZATION_SECURITY_CHECK_KEY, false)) {
-                    Object serializationType_obj = invocation.get(SERIALIZATION_ID_KEY);
-                    if (serializationType_obj != null) {
-                        if ((byte) serializationType_obj != serializationType) {
-                            throw new IOException("Unexpected serialization id:" + serializationType + " received from network, please check if the peer send the right id.");
+                if (invocation != null) {
+                    if (ConfigurationUtils.getSystemConfiguration().getBoolean(SERIALIZATION_SECURITY_CHECK_KEY, false)) {
```
Comment (changfubai) [inline dubbo-rpc/dubbo-rpc-dubbo/src/main/java/org/apache/dubbo/rpc/protocol/dubbo/DecodeableRpcResult.java:122]: I think it's better to check null after line 122

**Review by xiaoheng1 — APPROVED** [review]
LGTM

File: dubbo-rpc/dubbo-rpc-dubbo/src/main/java/org/apache/dubbo/rpc/protocol/dubbo/DecodeableRpcResult.java (hunk 1 above)
Comment (BazookaW) [reply dubbo-rpc/dubbo-rpc-dubbo/src/main/java/org/apache/dubbo/rpc/protocol/dubbo/DecodeableRpcResult.java:122]: I think when invocation is null, inside logic maybe unnecessary execute

Comment (horizonzy) [inline dubbo-rpc/dubbo-rpc-dubbo/src/main/java/org/apache/dubbo/rpc/protocol/dubbo/DecodeableRpcResult.java:122]: Hi, I notice that default value is true at pr #8547. Why here default value change to false.

Comment (BazookaW) [reply dubbo-rpc/dubbo-rpc-dubbo/src/main/java/org/apache/dubbo/rpc/protocol/dubbo/DecodeableRpcResult.java:122]: on branch 3.0, PR #8073 change default value to true

Comment (horizonzy) [reply dubbo-rpc/dubbo-rpc-dubbo/src/main/java/org/apache/dubbo/rpc/protocol/dubbo/DecodeableRpcResult.java:122]: @AlbumenJ pls help to check default value in master.

**Pull request merged**